content/productivity/custom-distribute-silver/plugin.js

```diff
@@ -0,0 +1,456 @@
+    });
+
+    // target display
+    const targetPlanetContainer = document.createElement("div");
+    targetPlanetContainer.innerText = "Current target: none";
+    // button for adding an target
+    const addPlanetTargetButton = document.createElement("button");
+    addPlanetTargetButton.innerText = "Add target";
+    addPlanetTargetButton.style.marginRight = "10px";
+    addPlanetTargetButton.addEventListener("click", () => {
+      removeAllChildNodes(targetPlanetContainer);
+      const targetedPlanet = ui.getSelectedPlanet();
+      if (targetedPlanet) {
+          this.planetTarget = targetedPlanet.locationId;
+      }
+      // make the planet either be "none" when nothing is selected, or the planet link.
+      targetPlanetContainer.append("Current target: ", targetedPlanet ? planetLink(targetedPlanet.locationId) : "none");
+    });
+
+    // button for test the pair (source => target)
+    const testButton = document.createElement("button");
+    testButton.innerText = "Test";
+    testButton.style.marginRight = "10px";
+    testButton.addEventListener("click", () => {
+      this.sendAndWithdraw(this.planetSource, this.planetTarget);
+    });
+
+    // addButton for append pair to the list
+    const addButton = document.createElement("button");
+    addButton.innerText = "AddList";
+    addButton.style.marginRight = "10px";
+    addButton.addEventListener("click", () => {
+      for (const item of this.pairList) {
+        if (item[0] == this.planetSource && item[1] == this.planetTarget) {
+          console.log('add duplicate pair, ignore it');
+          return;
+        }
+      }
+      const pair = [this.planetSource, this.planetTarget];
+      this.pairList.splice(0, 0, pair);
+      this.renderPairList(pairListContainer);
+    });
+
+    // pair list display
+    const pairListContainer = document.createElement("div");
+    pairListContainer.innerText = "Pair list: none";
+    pairListContainer.style.marginTop = '10px';
+
+    /*
+    // clearButton
+    const clearButton = document.createElement("button");
+    clearButton.innerText = "!!! ClearList !!!";
+    clearButton.style.marginRight = "10px";
+    clearButton.style.marginBottom = '10px';
+    clearButton.addEventListener("click", () => {
+      this.pairList = [];
+      this.renderPairList(pairListContainer);
+    });
+    */
+
+    // downloadButton
+    const downloadButton = document.createElement("button");
+    downloadButton.innerText = "Download Pairlist";
+    downloadButton.style.marginRight = "10px";
+    downloadButton.style.marginBottom = '10px';
+    downloadButton.addEventListener("click", () => {
+      let pairs = JSON.stringify(this.pairList);
+      var blob = new Blob([pairs], { type: 'application/json' }),
+      anchor = document.createElement('a');
+      anchor.download = df.getContractAddress().substring(0, 6) + '_pairs.json';
```

Comment:
anchor.download = new Date().toLocaleString() + "_" + df.getAccount() + '_Silver-pairs.json';

I like this format of output. More flexibile for more variants. Why is originaly df.getContractAddress?

Reply:
Thanks very much for your review and advise.
Output file with date can avoid duplicating the name when downloading.
`df.getAccount()` is more readability and flexibile than `df.getContractAddress()`.